cpp/include/cudf/strings/string_view.inl

```diff
@@ -262,17 +262,48 @@ __device__ inline int string_view::compare(const string_view& in) const
 
 __device__ inline int string_view::compare(const char* data, size_type bytes) const
 {
-  size_type const len1      = size_bytes();
-  const unsigned char* ptr1 = reinterpret_cast<const unsigned char*>(this->data());
-  const unsigned char* ptr2 = reinterpret_cast<const unsigned char*>(data);
-  size_type idx             = 0;
-  for (; (idx < len1) && (idx < bytes); ++idx) {
+  unsigned char const* ptr1 = reinterpret_cast<const unsigned char*>(this->data());
+  unsigned char const* ptr2 = reinterpret_cast<const unsigned char*>(data);
+
+  size_type const len1 = size_bytes();
+  size_type const len2 = bytes;
+  size_type const len  = std::min(len1, len2);
+  size_type idx        = 0;
+#if defined(__CUDA_ARCH__)
```

Comment:
When is `string_view::compare` used outside of device code?

Reply:
It is not but for some reason, the compiler fails on finding `__funnelshift_r` and `__byteperm` unless this is included.

Reply:
That doesn't sound right. We should look into it more.

Reply:
I don't want this change to hold up this PR which is intended to simply create a fastpath for single strings column sort. It appears that the `string_view::compare` optimization may become a much more involved change than I thought so I will move this code to a future PR instead.